### ISSUE TITLE
List of Neural Information Retrieval Models

### ISSUE THREAD

**Issue by rejasupotaro** [rejasupotaro/master-thesis#5]
## [2019] [HCAN](https://github.com/rejasupotaro/paper-reading/issues/110)

- Metrics: MAP, MRR, macro-F1, Acc, P@30
- Loss: Pointwise Cross-Entropy Loss
- Optimizer: SGD
- Implementation: https://github.com/jinfengr/hcan

## [2019] [CEDR](https://github.com/rejasupotaro/paper-reading/issues/84)

- Metrics: P@20, nDCG@20, ERR@20
- Loss: Pairwise Cross-Entropy Loss
- Optimizer: Adam
- Implementation: https://github.com/Georgetown-IR-Lab/cedr

## [2019] [Passage Re-ranking with BERT](https://github.com/rejasupotaro/paper-reading/issues/118)

- Metrics: MRR@10, MAP
- Loss: Pointwise Cross-Entropy Loss
- Implementation: https://github.com/nyu-dl/dl4marco-bert

## [2019] [DeepCT](https://github.com/rejasupotaro/paper-reading/issues/90)

- Metrics: MRR@10, MAP
- Loss: MSE to predict term weights
- Implementation: https://github.com/AdeDZY/DeepCT

## [2018] [SNRM](https://github.com/rejasupotaro/paper-reading/issues/18)

- Metrics: MAP, P@20, nDCD@20, Recall
- Loss: Pairwise Hinge Loss
- Optimizer: Adam
- Implementation: https://github.com/hamed-zamani/snrm

## [2018] Conv-KNRM

- Paper: https://dl.acm.org/doi/10.1145/3159652.3159659
- Metrics: nDCG@1, nDCG@10, MRR
- Loss: Pairwise Hinge Loss
- Optimizer: Adam
- Implementation:
  - https://github.com/NTMC-Community/MatchZoo
  - https://github.com/thunlp/Kernel-Based-Neural-Ranking-Models

## [2018] [NPRF](https://github.com/rejasupotaro/paper-reading/issues/67)

- Metrics: MAP, P@20, nDCG@20
- Loss: Pairwise Hinge Loss
- Optimizer: Adam
- Implementation: https://github.com/ucasir/NPRF

## [2018] [DRMM + PACRR](https://github.com/rejasupotaro/paper-reading/issues/22)

- Metrics: MAP, P@20, nDCG@20
- Loss: Pairwise Loss
- Implementation: https://github.com/nlpaueb/deep-relevance-ranking

## [2017] [NRM-F](https://github.com/rejasupotaro/paper-reading/issues/24)

- Metrics: nDCG@1, nDCG@10
- Loss: Pairwise Cross-Entropy Loss

## [2017] [End-to-End Neural Ad-hoc Ranking with Kernel Pooling](https://github.com/rejasupotaro/paper-reading/issues/113)

- Metrics: nDCG@1, nDCG@3, nDCG@10
- Loss: Pairwise Hinge Loss
- Implementation: 

## [2017] KNRM

- Paper: https://arxiv.org/abs/1904.12683
- Optimizer: Adam
- Implementation:
  - https://github.com/sebastian-hofstaetter/sigir19-neural-ir
  - https://github.com/NTMC-Community/MatchZoo
  - https://github.com/thunlp/Kernel-Based-Neural-Ranking-Models

## [2017] Co-PACRR

- Paper: https://arxiv.org/abs/1706.10192
- Implementation: 

## [2017] [PACRR](https://github.com/rejasupotaro/paper-reading/issues/50)

- Metrics: ERR@20, nDCG@20
- Loss: Pairwise Hinge Loss
- Implementation: 

## [2017] [DRMM](https://github.com/rejasupotaro/paper-reading/issues/35)

- Metrics: MAP, nDCG@20, P@20
- Loss: Pairwise Hinge Loss
- Implementation:
  - https://github.com/NTMC-Community/MatchZoo, 
  - https://github.com/faneshion/DRMM
  - https://github.com/nlpaueb/deep-relevance-ranking/tree/master/models/drmm
  - https://github.com/sebastian-hofstaetter/neural-ranking-drmm
  - https://github.com/X-Wei/ETH-thesis-TREC/tree/master/2-DRMM
  - https://github.com/felipemoraes/pyNeuIR

## [2017] DUET

- Paper: https://www.microsoft.com/en-us/research/publication/learning-match-using-local-distributed-representations-text-web-search/
- Loss: Pairwise Hinge Loss
- Metrics: nDCG@1, nDCG@10
- Implementation:
  - https://github.com/spacemanidol/MSMARCO/tree/master/Ranking/Baselines
  - https://github.com/NTMC-Community/MatchZoo
  - https://github.com/felipemoraes/pyNeuIR

Okapi BM25, Query Likelihood

## [2016] [DESM](https://github.com/rejasupotaro/paper-reading/issues/124)

- Metrics: nDCG@1, nDCG@3, nDCG@10

## [2014] CDSSM

- Paper: https://www.microsoft.com/en-us/research/publication/learning-semantic-representations-using-convolutional-neural-networks-for-web-search/
- Metrics: nDCG@1, nDCG@3, nDCG@10
- Implementation: https://github.com/NTMC-Community/MatchZoo

## [2013] [DSSM](https://github.com/rejasupotaro/paper-reading/issues/123)

- Metrics: nDCG@1, nDCG@3, nDCG@10
- Implementation: https://github.com/NTMC-Community/MatchZoo

## Metrics

https://github.com/rejasupotaro/paper-reading/issues/129

- Total gain
- Total time spent
- Stopping depth

## Others

- https://github.com/MaurizioFD/RecSys2019_DeepLearning_Evaluation
- https://github.com/haowei01/pytorch-examples
- https://github.com/mzhang001/tfranknet
- https://github.com/castorini/anserini
- https://github.com/teanalab
- https://github.com/github/CodeSearchNet
- https://github.com/QingyaoAi
